### PR TITLE
[SERVICE-417] Replace IAB usage with Channels API

### PR DESCRIPTION
**Pull request by ottah**
Client & Provider IAB calls changed to use channels instead.

### CODE REVIEW

File: src/provider/FDC3.ts [hunk 1]
```diff
@@ -360,27 +360,27 @@ export class FDC3 {
...
     }
 
-    private async sendContext(context: ContextBase): Promise<void> {
-        return fin.InterApplicationBus.publish('context', context);
+    private async sendContext(context: ContextBase): Promise<any> {
+        return this.apiHandler.channel.publish('context', context);
     }
 
-    private async sendIntent(intent: RaiseIntentPayload, targetApp: AppMetadata): Promise<void> {
+    private async sendIntent(intent: RaiseIntentPayload, targetApp: AppMetadata): Promise<any> {
```
Comment (bryangaleopenfin) [inline src/provider/FDC3.ts]: Do we want to change the signature? I feel like a change to our internal messaging system shouldn't change `void` to `any`. Could do this.apiHandler.channel.publish('intent', intent).then(() => {});

Comment (bryangaleopenfin) [reply src/provider/FDC3.ts]: Ditto for `sendContext`

File: src/provider/FDC3.ts [hunk 2]
```diff
@@ -254,7 +254,7 @@ export class FDC3 {
                             clearTimeout(timeout);
                             if (context) {
                                 // Pass context to application before resolving
-                                fin.InterApplicationBus.publish('context', {context}).then(resolve).catch((reason: string) => reject(new Error(reason)));
+                                this.apiHandler.channel.dispatch(app.identity, 'context', {context});
```
Comment (bryangaleopenfin) [inline src/provider/FDC3.ts]: Just want to check we're happy to lose the error handling here

**Review by bryangaleopenfin — COMMENTED** [review]
Looks good, just a couple of things I wanted to check

File: src/provider/FDC3.ts (hunk 2 above)
Comment (tomrobertsOF) [reply src/provider/FDC3.ts]: Not just the error handling, but the `.then(resolve)` was also removed. This is what makes sure we have sent the context before resolving the `open` call, and should definitely be there.

File: src/provider/FDC3.ts (hunk 1 above)
Comment (pjbroadbent) [reply src/provider/FDC3.ts]: +1 - if the return value isn't used, should keep the return type as `void`.

Methods are `async` already, replace the `return`s with `await`s?..

**Review by pjbroadbent — APPROVED** [review]

**Review by bryangaleopenfin — APPROVED** [review]

**Pull request merged**